packages/editor/src/lib/editor/managers/SnapManager.ts

```diff
@@ -151,27 +148,26 @@ function findAdjacentGaps(
 
 	const nextNodes = new Set<TLShapeId>()
 
-	for (const match of matches) {
+	matches.forEach((match) => {
 		const node = direction === 'forward' ? match.endNode.id : match.startNode.id
 		if (!nextNodes.has(node)) {
 			nextNodes.add(node)
-			matches.push(
-				...findAdjacentGaps(
-					gaps,
-					node,
-					gapLength,
-					direction,
-					rangeIntersection(
-						match.breadthIntersection[0],
-						match.breadthIntersection[1],
-						intersection[0],
-						intersection[1]
-					)!,
-					depth + 1
-				)
+			const foundGaps = findAdjacentGaps(
+				gaps,
+				node,
+				gapLength,
+				direction,
+				rangeIntersection(
+					match.breadthIntersection[0],
+					match.breadthIntersection[1],
+					intersection[0],
+					intersection[1]
+				)!
 			)
+
+			matches.push(...foundGaps)
```

Comment:
This was the critical change. Because we push into `matches`, the `for of` loop would run forever.

Reply:
ah good find!